packages/cli/src/commands/migrate/packageScripts.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { resolve as resolvePath } from 'path';
+import { PackageGraph } from '../../lib/monorepo';
+import { getRoleFromPackage, getRoleInfo, PackageRole } from '../../lib/role';
+
+const noStartRoles: PackageRole[] = ['cli', 'common-library'];
+
+export async function command() {
+  const packages = await PackageGraph.listTargetPackages();
+
+  await Promise.all(
+    packages.map(async ({ dir, packageJson }) => {
+      const role = getRoleFromPackage(packageJson);
+      if (!role) {
+        return;
+      }
+
+      const roleInfo = getRoleInfo(role);
+      const hasStart = !noStartRoles.includes(role);
+      const isBundled = roleInfo.output.includes('bundle');
+
+      const expectedScripts = {
+        ...(hasStart && { start: 'backstage-cli script start' }),
+        ...(isBundled
+          ? { bundle: 'backstage-cli script bundle', build: undefined }
+          : { build: 'backstage-cli script build', bundle: undefined }),
```

Comment:
are the undefined here to simplify the comparison below? looks a bit odd - do you rely on writeJson skipping the undefined keys then?